src/kudu/consensus/raft_consensus.cc

```diff
@@ -3661,12 +3662,20 @@ void RaftConsensus::HandleProxyRequest(const ConsensusRequestPB* request,
       // Ensure that the OpIds match. We don't expect a mismatch to ever
       // happen, so we log an error locally before reponding to the caller.
       if (!OpIdEquals(request->ops(i).id(), messages[i]->get()->id())) {
-          Status s = Status::IllegalState(Substitute(
-              "log cache returned non-consecutive OpId indexes: expected {}, found {}",
-              OpIdToString(request->ops(i).id()),
-              OpIdToString(messages[i]->get()->id())));
-          LOG_WITH_PREFIX(ERROR) << s.ToString();
-          RET_RESPOND_ERROR_NOT_OK(s);
+        string extra_info;
```

Comment:
although this change looks OK, we retrieve messages from log cache based on first_op_id - 1. We shouldn't
have to waste cycles to check this for proxy requests, as this should be more like a DEBUG check.

Reply:
This is not likely to be an expensive check, it's just doing two 64-bit integers compares.